axum-debug/src/lib.rs

```diff
@@ -84,21 +79,331 @@
+            return error;
+        }
+
+        if let Err(error) = param_limit_check(sig) {
+            return error;
+        }
+
+        let check_trait = check_trait_code(sig, &generics);
+        let check_return = check_return_code(sig, &generics);
+        let check_params = check_params_code(sig, &generics);
+
+        let expanded = quote_spanned! {span=>
+            #vis #sig {
+                #check_trait
+                #check_return
+                #(#check_params)*
+
+                #sig #block
+
+                #ident(#(#params),*).await
+            }
+        };
+
+        expanded.into()
+    }
+
+    fn create_generics(len: usize) -> Vec<Ident> {
+        let mut vec = Vec::new();
+        for i in 1..=len {
+            vec.push(Ident::new(&format!("T{}", i), Span::call_site()));
+        }
+        vec
+    }
+
+    fn async_check(sig: &Signature) -> Result<(), TokenStream> {
+        if sig.asyncness.is_none() {
+            let error = syn::Error::new_spanned(sig.fn_token, "handlers must be async functions")
+                .to_compile_error()
+                .into();
+
+            return Err(error);
+        }
+
+        Ok(())
+    }
+
+    fn param_limit_check(sig: &Signature) -> Result<(), TokenStream> {
+        if sig.inputs.len() > 16 {
+            let msg = "too many extractors. 16 extractors are allowed\n\
+                       note: you can nest extractors like \"a: (Extractor, Extractor), b: (Extractor, Extractor)\"";
+
+            let error = syn::Error::new_spanned(&sig.inputs, msg)
+                .to_compile_error()
+                .into();
+
+            return Err(error);
+        }
+
+        Ok(())
+    }
+
+    fn check_trait_code(sig: &Signature, generics: &[Ident]) -> proc_macro2::TokenStream {
+        let ident = &sig.ident;
+        let span = ident.span();
+
+        quote_spanned! {span=>
+            {
+                debug_handler(#ident);
+
+                fn debug_handler<F, Fut, #(#generics),*>(_f: F)
+                where
+                    F: ::std::ops::FnOnce(#(#generics),*) -> Fut + Clone + Send + Sync + 'static,
+                    Fut: ::std::future::Future + Send,
+                {}
+            }
+        }
+    }
+
+    fn check_return_code(sig: &Signature, generics: &[Ident]) -> proc_macro2::TokenStream {
+        let span = match &sig.output {
+            ReturnType::Default => syn::Error::new_spanned(&sig.output, "").span(),
+            ReturnType::Type(_, t) => syn::Error::new_spanned(t, "").span(),
+        };
+        let ident = &sig.ident;
+
+        quote_spanned! {span=>
+            {
+                debug_handler(#ident);
+
+                fn debug_handler<F, Fut, Res, #(#generics),*>(_f: F)
+                where
+                    F: ::std::ops::FnOnce(#(#generics),*) -> Fut,
+                    Fut: ::std::future::Future<Output = Res>,
+                    Res: ::axum_debug::axum::response::IntoResponse,
```

Comment:
This needs to be `::axum::response::IntoResponse`.

Reply:
Fixed in 50b841b0f8a16

Reply:
Gonna add some tests in the next PR